### PR TITLE
Don't hardcode subnet and worker amounts

### PR DESCRIPTION
**Pull request by duboisph**
This PR adds some small changes:
- Creates master, worker and utility subnets according to available AZs
- Aloww to override the amount of subnets used by the worker instancegroup (eg. to deploy in a single AZ)
- Allows specifying the min amount of worker instances (eg. to to work in conjunction with the cluster-autoscaler)

Tested on a new stack

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Creates a full `kops` cluster specification yaml, including the required instanc
 * [`k8s_data_bucket`]: String(required): The S3 bucket that `kops` will use to store it's configuration and state.
 * [`k8s_version`]: String(required): The Kubernetes version to deploy.
 * [`vpc_id`]: String(required): The VPC in which the Kubernetes cluster must be deployed
+* [`min_amount_workers`]: String(required): the minimum amount of workers. Will default to the amount of AZs.
```
Comment (SamClinckspoor) [inline README.md]: if there is a default is it required?

File: cluster/main.tf [hunk 2]
```diff
@@ -121,7 +121,7 @@ data template_file "worker-instancegroup-spec" {
     kubernetes_ami              = "${data.aws_ami.kubernetes_ami.name}"
     subnets                     = "${join("\n", formatlist("  - worker-%s", data.aws_availability_zones.available.names))}"
     instance_type               = "${var.worker_instance_type}"
-    min                         = "${length(data.aws_availability_zones.available.names)}"
+    min                         = "${var.min_amount_workers != 0 ? var.min_amount_workers : length(data.aws_availability_zones.available.names)}"
```
Comment (SamClinckspoor) [inline cluster/main.tf]: maybe `>` is better than `!=` to prevent negative values

File: README.md (hunk 1 above)
Comment (venturel) [reply README.md]: Shouldn't be required, idd

Comment (SamClinckspoor) [reply README.md]: should also be an int instead of string

**Pull request merged**